AK/FloatingPointStringConversions.cpp

```diff
@@ -1589,75 +1575,45 @@ class MinimalBigInt {
 
     void multiply_with_power_of_2(u32 exponent)
     {
-        // It's cheaper to shift bits first since that creates at most 1 new word
-        shift_bits(exponent % 64);
-        shift_words(exponent / 64);
+        if (exponent) {
+            size_t max_new_length = m_used_length + (exponent + word_size - 1) / word_size;
+            if (m_used_length != max_words_needed)
+                m_words[m_used_length] = 0;
+            auto storage = get_storage(max_new_length);
+
+            StorageOperations::shift_left(storage, exponent, storage);
+            trim_last_word_if_zero();
+        }
     }
 
     enum class CompareResult {
-        Equal,
-        GreaterThan,
-        LessThan
+        Equal = 0,
+        GreaterThan = 1,
+        LessThan = -1
     };
 
     CompareResult compare_to(MinimalBigInt const& other) const
     {
-        if (m_used_length > other.m_used_length)
-            return CompareResult::GreaterThan;
-
-        if (m_used_length < other.m_used_length)
-            return CompareResult::LessThan;
-
-        // Now we know it's the same size
-        for (size_t i = m_used_length; i > 0; --i) {
-            auto our_word = m_words[i - 1];
-            auto their_word = other.m_words[i - 1];
-
-            if (our_word > their_word)
-                return CompareResult::GreaterThan;
-            if (their_word > our_word)
-                return CompareResult::LessThan;
-        }
-
-        return CompareResult::Equal;
+        return static_cast<CompareResult>(StorageOperations::compare(get_storage(), other.get_storage(), false));
     }
 
 private:
-    void shift_words(u32 amount)
+    UnsignedStorageSpan get_storage(size_t new_length = 0)
     {
-        if (amount == 0)
-            return;
-
-        VERIFY(amount + m_used_length <= max_words_needed);
-
-        for (size_t i = m_used_length + amount - 1; i > amount - 1; --i)
-            m_words[i] = m_words[i - amount];
-
-        for (size_t i = 0; i < amount; ++i)
-            m_words[i] = 0;
-
-        m_used_length += amount;
+        if (new_length > m_used_length)
+            m_used_length = min(max_words_needed, new_length);
+        return { m_words.data(), m_used_length };
     }
 
-    void shift_bits(u32 amount)
+    AK::UnsignedStorageReadonlySpan get_storage() const
```

Comment:
Why is this function here? Won't it always use the other one since it has a default parameter value?

Also if this should be kept the `AK` prefix should be removed

Reply:
Ah I missed the const